### PR TITLE
Fit Grains

### PR DESCRIPTION
**Pull request by johnkit**
Adds dialog and logic for running the fit-grains code.
* Displays a fit-grains dialog when the indexing process is completed
* Displays a tabular view of the "grains table" results when fit-grains is completed
* Updates processes to run in the background with progress dialogs for (i) generating eta-omega maps, (ii) running indexing, and (iii) running fit-grains.

Notes:
* Only successfully completes the fit-grains processing when the image data were originally loaded from the hexrdgui menu (not the "Load Data" tab)
* This PR is dependent on [hexrd PR 62](https://github.com/HEXRD/hexrd/pull/62)

### CODE REVIEW

**Comment by cjh1** [comment]
@johnkit I am getting the following exception when I try to run the grain fitting:

```
Traceback (most recent call last):
  File "/home/cjh/work/source/hexrdgui/hexrd/ui/indexing/fit_grains_dialog.py", line 75, in on_accepted
    self.update_config()
  File "/home/cjh/work/source/hexrdgui/hexrd/ui/indexing/fit_grains_dialog.py", line 157, in update_config
    config['refit'][1] = self.ui.refit_ome_scale.value()
AttributeError: 'PySide2.QtWidgets.QDialog' object has no attribute 'refit_ome_scale'

```

File: hexrd/ui/indexing/run.py [hunk 1]
```diff
@@ -141,15 +130,16 @@ def run_grain_fitting(self):
...
         print('Grain fitting is complete!')
         print(f'{self.qbar.shape[1]} grains were found')
 
-    def fit_grains_config_finished(self, result):
-        # result 0 == rejected, 1 == accepted
-        if result == 1:
-            from PySide2.QtWidgets import QMessageBox
-            QMessageBox.information(
-                None, "Todo", "Sorry, Fit Grains not yet implemented")
+        from PySide2.QtWidgets import QMessageBox
```
Comment (cjh1) [inline hexrd/ui/indexing/run.py:143]: Is there a particular reason to do the import here? Otherwise lets move it to the top.

Comment (johnkit) [reply hexrd/ui/indexing/run.py:143]: Nope, no good reason for putting the import here. I'll force push a change to move it to the top.

File: hexrd/ui/indexing/fit_grains_options_dialog.py [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+class FitGrainsOptionsDialog(QObject):
+    accepted = Signal()
+    rejected = Signal()
+
+    def __init__(self, parent=None):
+        super().__init__(parent)
+
+        config = HexrdConfig().indexing_config['fit_grains']
+        if config.get('do_fit') is False:
+            return
```
Comment (cjh1) [inline hexrd/ui/indexing/fit_grains_options_dialog.py:21]: Left over debug?

File: hexrd/ui/indexing/fit_grains_results_model.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+    model = FitGrainsResultsModel(data)
+    view = QTableView()
+    view.verticalHeader().hide()
+    view.setModel(model)
+    view.resizeColumnToContents(0)
+    view.resize(960, 320)
+
+    view.show()
+    app.exec_()
```
Comment (cjh1) [inline hexrd/ui/indexing/fit_grains_results_model.py:78]: Leftover debug?

**Comment by joelvbernier** [comment]
@johnkit do we want to resolve @cjh1's questions before merging up?

**Comment by johnkit** [comment]
> @johnkit do we want to resolve @cjh1's questions before merging up?

Sorry Joel, I've been offline most of the day. Looks like Chris is offline, at least at the moment.

If we're holding you up, I can go ahead and merge. I would like to keep the code that Chris suggests isn't needed, but we can always remove it later. His calendar shows him in a meeting until 3pm. If he's not online or we can't resolve this by, let's say 3:15, then I will go ahead and merge as is.

**Comment by johnkit** [comment]
@joelvbernier  In case you didn't get the email, it just got merged now.

**Comment by cjh1** [comment]
Sorry we had a power cut here, its been hard to get online ...

**Pull request merged**